cpp/src/backends/torch_scripted/handler/base_handler.hh

```diff
@@ -70,12 +74,38 @@ class BaseHandler {
       std::shared_ptr<torchserve::InferenceRequestBatch>& request_batch,
       std::shared_ptr<torchserve::InferenceResponseBatch>& response_batch) {
     try {
+      auto start_time = std::chrono::high_resolution_clock::now();
+
       std::map<uint8_t, std::string> idx_to_req_id;
       auto inputs =
           Preprocess(device, idx_to_req_id, request_batch, response_batch);
       auto outputs =
           Inference(model, inputs, device, idx_to_req_id, response_batch);
       Postprocess(outputs, idx_to_req_id, response_batch);
+
+      auto stop_time = std::chrono::high_resolution_clock::now();
+      std::chrono::duration<double, std::milli> duration =
+          stop_time - start_time;
+      try {
+        auto& handler_time_metric =
+            torchserve::MetricsRegistry::GetMetricsCacheInstance()->GetMetric(
+                torchserve::MetricType::GAUGE, "HandlerTime");
+        std::string handler_time_metric_request_id = "";
+        for (auto request_iter = request_batch->begin();
+             request_iter != request_batch->end(); request_iter++) {
+          handler_time_metric_request_id += request_iter->request_id;
+          if (std::next(request_iter) != request_batch->end()) {
+            handler_time_metric_request_id += ",";
+          }
+        }
```

Comment:
It is not efficient to create request id string in each metrics log. the request ids string can be created at the preprocess once, and used in any metrics log. check https://github.com/pytorch/serve/blob/cpp_backend/cpp/src/backends/torch_scripted/handler/base_handler.cc#L47